### DISCUSSION TITLE
Merge & Simplify Occluder Polygon for TileMaps

### DISCUSSION THREAD

**Discussion by 8uurg** [godotengine/godot-proposals#7934] [2D]
During the development of my first (2D) game with Godot, I experienced a large framerate drop with a few Light2D's & many small tiles with a full-tile (square) occluder polygon specified. If either the shadow casting or the occluders are disabled, no framerate drop is present. As a workaround I've manually created a polygon that covers many tiles at once, which provides the same result without the framerate drop - at the cost of manually having to create LightOccluder2D's and their corresponding polygons.

As this is quite tedious, and gets out of sync - especially if the tilemap is changed dynamically - it would be appreciated if either the TileMap or the algorithm used to determine shadows could perform this optimization automatically.

**Comment by Calinou** [comment]
The 2D navmesh/collision baking from https://github.com/godotengine/godot/pull/69988 should be usable for light occluders as well.

**Reply by smix8** [reply]
I think Calinou wanted to link https://github.com/godotengine/godot/pull/80796.

And yes in general it can be used to bake any kind of polygons and the main problem is the missing Editor interface for it.
